src/test/kotlin/app/batch/KeyPairGeneratorTest.kt

```diff
@@ -0,0 +1,142 @@
+                KeyPair("cdb.collection.0000.json.gz.enc", "cdb.collection.0000.json.gz.encryption.json"),
+                KeyPair("sdb.collection.0000.json.gz.enc", "sdb.collection.0000.json.gz.encryption.json"))
+
+        val keys = listOf("adb.collection.0000.json.gz.enc", "cdb.collection.0000.json.gz.enc", "sdb.collection.0000.json.gz.enc",
+                "adb.collection.0000.json.gz.encryption.json", "cdb.collection.0000.json.gz.encryption.json",
+                "sdb.collection.0000.json.gz.encryption.json", "abc.json", "edf.json")
+
+        val actualKeyPairs = keyPairGenerator.generateKeyPairs(keys, fileFormatRegex, dataFileExtensionRegex, metadataFileExtensionRegex)
+
+        val captor = argumentCaptor<ILoggingEvent>()
+        verify(mockAppender, times(8)).doAppend(captor.capture())
+        val formattedMessages = captor.allValues.map { it.formattedMessage }
+        assertTrue(formattedMessages.contains("2 key(s) that don't match the given file fileFormat $fileFormatRegex found"))
+        assertTrue(formattedMessages.contains("Unmatched keys : abc.json,edf.json"))
+        assertEquals(expectedKeyPairs, actualKeyPairs)
+
+    }
+
+    @Test
+    fun Should_Log_When_Given_Keys_Dont_Match_File_Extensions() {
+
+        val root = LoggerFactory.getLogger(ch.qos.logback.classic.Logger.ROOT_LOGGER_NAME) as ch.qos.logback.classic.Logger
+        val mockAppender: Appender<ILoggingEvent> = mock()
+        root.addAppender(mockAppender)
+        val expectedKeyPairs = listOf(KeyPair("adb.collection.0000.json.gz.enc", "adb.collection.0000.json.gz.encryption.json"),
+                KeyPair("cdb.collection.0000.json.gz.enc", "cdb.collection.0000.json.gz.encryption.json"),
+                KeyPair("sdb.collection.0000.json.gz.enc", "sdb.collection.0000.json.gz.encryption.json"))
+
+        val keys = listOf("adb.collection.0000.json.gz.enc", "cdb.collection.0000.json.gz.enc", "sdb.collection.0000.json.gz.enc",
+                "adb.collection.0000.json.gz.encryption.json", "cdb.collection.0000.json.gz.encryption.json",
+                "sdb.collection.0000.json.gz.encryption.json", "adb.collection.0000.json.gz.pdf", "adb.collection.0000.json.gz.docx")
+
+        val actualKeyPairs = keyPairGenerator.generateKeyPairs(keys, fileFormatRegex, dataFileExtensionRegex, metadataFileExtensionRegex)
+
+        val captor = argumentCaptor<ILoggingEvent>()
+        verify(mockAppender, times(8)).doAppend(captor.capture())
+        val formattedMessages = captor.allValues.map { it.formattedMessage }
+        assertTrue(formattedMessages.contains("adb.collection.0000.json.gz.pdf,adb.collection.0000.json.gz.docx matched file format but not matched neither data nor metadata file extension"))
+        assertEquals(expectedKeyPairs, actualKeyPairs)
+
+    }
+
+    @Test
+    fun Should_Log_When_Data_File_Doesnt_But_Metadata_File_Exist() {
```

Comment:
I suggest "Should_Log_When_Data_File_Doesn_Exist_But_Metadata_File_Does" or "Should_Log_When_Data_File_Doesnt_But_Metadata_File_Does_Exist" ??

Reply:
Modified to "Should_Log_When_Data_File_Doesnt_But_Metadata_File_Does_Exist" now